rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -56,31 +58,53 @@ SequentialWriter::SequentialWriter(
   storage_(nullptr),
   metadata_io_(std::move(metadata_io)),
   converter_(nullptr),
+  compressor_{nullptr},
   max_bagfile_size_(rosbag2_storage::storage_interfaces::MAX_BAGFILE_SIZE_NO_SPLIT),
   topics_names_to_info_(),
-  metadata_()
-{}
+  metadata_(),
+  compression_mode_{CompressionMode::NONE} {}
+
 
 SequentialWriter::~SequentialWriter()
 {
   reset();
 }
 
-void SequentialWriter::init_metadata()
+void SequentialWriter::init_compression(const CompressionOptions & compression_options)
+{
+  if (compression_options.compression_mode != rosbag2_cpp::CompressionMode::NONE) {
+    if (compression_options.compression_format == "zstd") {
+      compressor_ = std::make_unique<rosbag2_compression::ZstdCompressor>();
+    } else {
+      std::stringstream err;
+      err << "Unsupported compression format " << compression_options.compression_format;
+      throw std::runtime_error{err.str()};
```

Comment:
```suggestion
      throw std::invalid_argument{err.str()};
```

To be consistent with the changes below.